tools/chocolateypkg/chocolateypkg.go

```diff
@@ -94,33 +93,37 @@ func replaceNuspec(dir, version string) error {
 	return err
 }
 
-func generateSha256(url string) (string, error) {
-	r, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, http.NoBody)
+func sha256sum(path string) (string, error) {
+	f, err := os.Open(path)
 	if err != nil {
 		return "", err
 	}
-	resp, err := http.DefaultClient.Do(r)
-	if err != nil {
+	defer f.Close()
+
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
 		return "", err
 	}
 
-	defer resp.Body.Close()
-
-	h := sha256.New()
-	if _, err := io.Copy(h, resp.Body); err != nil {
+	if _, err := f.Stat(); err != nil {
 		return "", err
 	}
+
 	return hex.EncodeToString(h.Sum(nil)), nil
 }
 
 //go:embed chocolateyinstall.ps1
 var installScript string
 
-func replaceInstallScript(dir, url string) error {
-	checkSum, err := generateSha256(url)
+func generateInstallScript(dir, version string) error {
+	file := fmt.Sprintf("mongodb-atlas-cli_%s_windows_x86_64.msi", version)
+	checkSum, err := sha256sum(path.Join(dir, file))
 	if err != nil {
 		return err
 	}
+	// For previews this link will be invalid
+	url := fmt.Sprintf("https://fastdl.mongodb.org/mongocli/mongodb-atlas-cli_%s_windows_x86_64.msi", version)
```

Comment:
do we care? I tried to have point to our internal previews but it becomes complex as I need access to more evg variables for this to work

Reply:
I personally think we should not bother about this on preview as we do/should not publish to chocolatey on non final versions.